bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/generic/converter/ColorChannelHandler.java

```diff
@@ -77,32 +77,29 @@ public String toString(Command command) {
     public Optional<State> toState(String string) {
         State newState = UnDefType.UNDEF;
         if (string.equals(channelConfig.onValue)) {
-            if (state instanceof HSBType) {
-                newState = new HSBType(((HSBType) state).getHue(), ((HSBType) state).getSaturation(),
-                        PercentType.HUNDRED);
+            if (state instanceof HSBType hsbType) {
```

Comment:
Perhaps this would be slightly better for designating it as a state of a specific type?
```suggestion
            if (state instanceof HSBType hsbState) {
```

(same below)